File: Data Structures & Algorithms/longest-consecutive-sequence/submission-2.py
class Solution:
    def longestConsecutive(self, nums: List[int]) -> int:
        newset=set()
        res=0
        for i in nums:
            newset.add(i)
        for i in nums:
            if i-1 not in newset:
                count=1
                val=i+1
                while val in newset:
                    count+=1
                    val+=1
                res=max(res,count)
        
        return res
                 

File: Data Structures & Algorithms/longest-substring-without-duplicates/submission-2.cpp
class Solution {
public:
    int lengthOfLongestSubstring(string s) {
        int maxlen=0;
        for(int i=0;i<s.size();i++)
        {
            for(int j=i;j<s.size();j++)
            {
                string sub=s.substr(i,j-i+1);
                if(checkduplicates(sub))
                {
                    break;
                }
                else
                {
                    maxlen=max(maxlen,j-i+1);
                }
            }
        }
        return maxlen;
    }
    bool checkduplicates(string sub)
    {
        unordered_set<char> uset;
        for(char ch:sub)
        {
            if(uset.count(ch)) return true;
            uset.insert(ch);
        }
        return false;
    }
};
